modular_skyrat/modules/assault_operatives/code/assault_operatives.dm

```diff
@@ -0,0 +1,260 @@
+
+	human_target.equipOutfit(chosen_loadout_type)
+
+	human_target.regenerate_icons()
+
+	to_chat(human_target, loadout_desc)
+
+	equipped_class = chosen_loadout
+	return TRUE
+
+/datum/antagonist/assault_operative/proc/equip_operative()
+	if(!ishuman(owner.current))
+		return
+	var/mob/living/carbon/human/human_target = owner.current
+	for(var/obj/item/item in human_target.get_equipped_items(TRUE))
+		qdel(item)
+
+	var/obj/item/organ/brain/human_brain = human_target.getorganslot(BRAIN)
+	human_brain.destroy_all_skillchips() // get rid of skillchips to prevent runtimes
+
+	human_target.equipOutfit(assault_operative_default_outfit)
+
+	human_target.regenerate_icons()
+
+	return TRUE
+
+/datum/antagonist/assault_operative/create_team(datum/team/assault_operatives/new_team)
+	if(!new_team)
+		if(!always_new_team)
+			for(var/datum/antagonist/assault_operative/assault_operative in GLOB.antagonists)
+				if(!assault_operative.owner)
+					stack_trace("Antagonist datum without owner in GLOB.antagonists: [assault_operative]")
+					continue
+				if(assault_operative.assault_team)
+					assault_team = assault_operative.assault_team
+					return
+		assault_team = new /datum/team/assault_operatives
+		assault_team.add_member(owner)
+		assault_team.update_objectives()
+		return
+	if(!istype(new_team))
+		stack_trace("Wrong team type passed to [type] initialization.")
+	assault_team = new_team
+	assault_team.add_member(owner)
+	assault_team.update_objectives()
+
+
+/datum/antagonist/assault_operative/proc/move_to_spawnpoint()
+	var/team_number = 1
+	if(assault_team)
+		team_number = assault_team.members.Find(owner)
+	owner.current.forceMove(GLOB.assault_operative_start[((team_number - 1) % GLOB.assault_operative_start.len) + 1])
+
+/datum/antagonist/assault_operative/get_preview_icon()
+	if (!preview_outfit)
+		return null
+
+	var/icon/final_icon = render_preview_outfit(preview_outfit)
+
+	return finish_preview_icon(final_icon)
+
+/**
+ * ASSAULT OPERATIVE TEAM DATUM
+ */
+
+/datum/team/assault_operatives
+
+/datum/team/assault_operatives/proc/update_objectives(initial = FALSE)
+	var/untracked_heads = SSjob.get_all_heads()
+
+	for(var/datum/objective/interrogate/objective in objectives)
+		untracked_heads -= objective.target
+
+	for(var/datum/mind/mind in untracked_heads)
+		var/datum/objective/interrogate/new_target = new()
+		new_target.team = src
+		new_target.target = mind
+		new_target.update_explanation_text()
+		objectives += new_target
+
+	for(var/datum/mind/mind in members)
+		var/datum/antagonist/assault_operative/assault_op = mind.has_antag_datum(/datum/antagonist/assault_operative)
+		assault_op.objectives |= objectives
+
+	addtimer(CALLBACK(src,.proc/update_objectives), HEAD_UPDATE_PERIOD, TIMER_UNIQUE)
+
+/datum/team/assault_operatives/proc/operatives_dead()
+	for(var/i in members)
+		var/datum/mind/operative_mind = i
+		if(ishuman(operative_mind.current) && (operative_mind.current.stat != DEAD))
+			return FALSE
+	return TRUE
```

Comment:
More descriptive variable names than that, please. Plus, it might be a good idea to use the type of the variable there instead, and use as anything, but I don't remember if mind datums are fickle like clients, so I'd have to check first.